OpenRA.Mods.Common/Widgets/SupportPowerTimerWidget.cs

```diff
@@ -21,23 +21,45 @@ public class SupportPowerTimerWidget : Widget
 	{
 		public readonly string Font = "Bold";
 		public readonly string Format = "{0}: {1}";
-		public readonly TimerOrder Order = TimerOrder.Descending;
+		[Translate] public readonly string AlliesHeader = "Allies";
+		[Translate] public readonly string EnemiesHeader = "Enemies";
+		public enum TimerOrder { AlliesFirst, EnemiesFirst }
+		public readonly TimerOrder Order = TimerOrder.EnemiesFirst;
 
 		readonly IEnumerable<SupportPowerInstance> powers;
-		Pair<string, Color>[] texts;
+		readonly IEnumerable<SupportPowerInstance> powersAllies;
+		readonly IEnumerable<SupportPowerInstance> powersEnemies;
+		readonly IEnumerable<Player> playersAllies;
+		readonly IEnumerable<Player> playersEnemies;
+		Pair<string, Color>[] textsAllies;
+		Pair<string, Color>[] textsEnemies;
 
 		[ObjectCreator.UseCtor]
 		public SupportPowerTimerWidget(World world)
 		{
 			powers = world.ActorsWithTrait<SupportPowerManager>()
-				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant)
+				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant && p.Actor.Owner != world.LocalPlayer)
 				.SelectMany(s => s.Trait.Powers.Values)
 				.Where(p => p.Instances.Any() && p.Info.DisplayTimer && !p.Disabled);
+
+			powersAllies = powers.Where(p => world.LocalPlayer.IsAlliedWith(p.Instances[0].Self.Owner));
```

Comment:
This causes an NRE in replays. You may want to use `RenderPlayer` instead, but note that that too can be `null` in replays.